File: 415_add_strings.py
class Solution(object):
    def addStrings(self, num1, num2):
        num_dict = {'0':0, '1':1, '2':2, '3':3, '4':4, '5':5,  '6':6, '7':7, '8':8, '9':9}
        # num3 = 0
        
        # Covert
        def str_to_int(num_str):

            num = 0

            for char in num_str:
                num = 10 * num + num_dict[char]

            return num
        

        num1 = str_to_int(num1)

        num2 = str_to_int(num2)

        # num3 = num1 + num2
        
        return str(num1 + num2)

File: 415_add_strings.cpp
#include<string>
class Solution {
public:
    string addStrings(string num1, string num2) {
    int i = num1.size() - 1;
    int j = num2.size() - 1;
    int sum = 0;
    string res = "";
    while (i >= 0 || j >= 0 || sum) {
        if (i >= 0 && j >= 0) { sum += (num1[i--] - '0') + (num2[j--] - '0'); }
        else if (i >= 0) { sum += num1[i--] - '0'; }
        else if (j >= 0) { sum += num2[j--] - '0'; }
        res += sum % 10 + '0';
        sum /= 10;
    }
    reverse(begin(res), end(res));
    return res;
}
};